React/Modules/RCTServerUtils.m

```diff
@@ -0,0 +1,21 @@
+
+#import "RCTServerUtils.h"
+#import "RCTIPAddress.h"
+
+@implementation RCTServerUtils
+
++ (NSURL *)getServerURL:(NSString *)path
+{
+  NSUserDefaults *standardDefaults = [NSUserDefaults standardUserDefaults];
+  NSInteger port = [standardDefaults integerForKey:@"websocket-executor-port"] ?: 8081;
+
+#if TARGET_OS_SIMULATOR
```

Comment:
Let's not let the logic live across different files / build steps. Instead, put the condition into shell script (Xcode sets proper env vars)